docs/rules/no-restricted-syntax.md

```diff
@@ -12,7 +12,7 @@ This rule disallows specified (that is, user-defined) syntax.
 
 ## Options
 
-This rule takes a list of strings:
+This rule takes a list of strings, where each string is a node type or AST selector:
```

Comment:
Nitpick: Technically, any node type is also an AST selector, so this sentence is redundant. However, I'm fine with keeping it if you think it makes things clearer.